packages/procaptcha-common/src/state.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021-2024 Prosopo (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import {
+    Account,
+    CaptchaResponseBody,
+    ProcaptchaState,
+    ProcaptchaStateUpdateFn,
+    ProsopoCaptchaApiInterface,
+    TCaptchaSubmitResult,
+} from '@prosopo/types'
+
+type useRefType = <T>(defaultValue: T) => { current: T }
+type useStateType = <T>(defaultValue: T) => [T, (value: T) => void]
+
+export const buildUpdateState = (state: ProcaptchaState, onStateUpdate: ProcaptchaStateUpdateFn) => {
```

Comment:
This can just be

```
export const buildUpdateState = (state: ProcaptchaState, onStateUpdate: ProcaptchaStateUpdateFn) => 
    (nextState: Partial<ProcaptchaState>) => {
        Object.assign(state, nextState);
        onStateUpdate(nextState);
    };
```

Also quite noisy comments here